File: trunk/ServerPhp/src/dc/DC0091.php
<?php
/* N21 eBusiness Suite
 * Copyright: Nan21 Electronics srl
 * Generated content.
 * DC0091 Controller: Oracle sessions
 */

require_once(PATH_N21LIB."/Http/Controller.php");

class DC0091 extends Controller {

  public function __construct($request, $response, $session) {
    parent::__construct($request, $response, $session);
    $this->_initFields();
  }

private function preQuery(&$params, &$where) {
    if (!empty($_REQUEST["QRY_SQL_TEXT"])) {
      $where .= (!empty($where))?" and ":"";
      $where .= "t.SQL_TEXT like :SQL_TEXT";
      $params["SQL_TEXT"] = $_REQUEST["QRY_SQL_TEXT"];
    }
}

public function doQuery() {
    $start = nvl($this->getRequestParam("start"), 0);
    $limit = nvl($this->getRequestParam("limit"),40);
    $orderBy = (!empty($_REQUEST["sort"]))?$_REQUEST["sort"]:"";
    $orderSense = (!empty($_REQUEST["dir"]))?$_REQUEST["dir"]:"";
    $orderByClause = (!empty($orderBy))? "order by $orderBy $orderSense " : "" ;
    $where = "type='USER'";
    $params = array();
    $this->preQuery($params, $where);
    if (!empty($where)) {
      $where = " where ".$where;
    }
    $sql = "select 
                t.CLIENT_INFO
                ,t.LOGON_TIME
                ,t.MACHINE
                ,t.MODULE
                ,t.OSUSER
                ,t.PROCESS
                ,t.PROGRAM
                ,t.SADDR
                ,t.SCHEMANAME
                ,t.SERIAL_
                ,t.SERVER
                ,t.SERVICE_NAME
                ,t.SID
                ,t.SQL_ADDRESS
                ,t.SQL_ID
                ,t.SQL_TEXT
                ,t.STATE
                ,t.STATUS
                ,t.TERMINAL
                ,t.TYPE
                ,t.USERNAME
            from V_ORCL_SESSION t $where $orderByClause ";
    $this->logger->debug($sql);
    $rs = $this->getDbConn()->SelectLimit($sql, $limit, $start, $params);
    $rsCount = $this->getDbConn()->Execute("select count(*) TOTALCOUNT from (".$sql.") t", $params);
    $rsCount->MoveFirst();
    $dataOut = $this->serializeCursor($rs,array_keys($this->fields), $this->getDataFormatFetch());
    if ($this->getDataFormatFetch() == "xml" ) {header("Content-type: application/xml");}
    print $this->formatQueryResponseData($dataOut, $rsCount->fields["TOTALCOUNT"]);
} /* end function doQuery  */


public function doExport() {
    $start = nvl($this->getRequestParam("start"), 0);
    $limit = nvl($this->getRequestParam("limit"),40);
    $groupBy = (!empty($_REQUEST["groupBy"]))?$_REQUEST["groupBy"]:"";
    $orderBy = (!empty($_REQUEST["sort"]))?$_REQUEST["sort"]:"";
    $orderSense = (!empty($_REQUEST["dir"]))?$_REQUEST["dir"]:"";
    $orderByClause = (!empty($orderBy))? "order by $orderBy $orderSense " : "" ;
    $where = "";
    $params = array();
    $this->preQuery($params, $where);
    if (!empty($where)) {
      $where = " where ".$where;
    }
    $sql = "select 
                t.SID
                ,t.SERIAL_
                ,t.USERNAME
                ,t.STATUS
                ,t.MACHINE
                ,t.TERMINAL
                ,t.OSUSER
                ,t.PROGRAM
                ,t.TYPE
                ,t.LOGON_TIME
                ,t.SADDR
                ,t.SERVER
                ,t.SCHEMANAME
                ,t.PROCESS
                ,t.SQL_ADDRESS
                ,t.SQL_ID
                ,t.MODULE
                ,t.CLIENT_INFO
                ,t.STATE
                ,t.SERVICE_NAME
                ,t.SQL_TEXT
            from V_ORCL_SESSION t $where $orderByClause ";
    $rs = $this->getDbConn()->Execute($sql, $params);
    $rsCount = $this->getDbConn()->Execute("select count(*) TOTALCOUNT from (".$sql.") t", $params);
    $rsCount->MoveFirst();
    if (!empty($_REQUEST["_p_disp_cols"])) {
      $columns = explode("|",$_REQUEST["_p_disp_cols"]);
    } else {
      $columns = array_keys($this->fields);
    }
    if ($this->getDataFormatExport() == "csv" ) {
      $dataOut = $this->serializeCursor($rs,$columns,"csv");
    } else {
      $dataOut = $this->serializeCursor($rs,$columns,"xml");
      $dataOut = "<records>".$dataOut."</records>";
      $dataOut = "<queryParams>".$this->serializeMap($params,"xml")."</queryParams>".$dataOut;
      $dataOut = "<columnDef>".$this->columnDefForExport($columns,$this->fields,true).$this->columnDefForExport(array_diff(array_keys($params), $columns),$this->fields,false)."</columnDef>".$dataOut;
      $dataOut = "<staticText>".$this->exportLocalizedStaticText()."</staticText>".$dataOut;
      $dataOut = "<groupBy>".$groupBy."</groupBy>".$dataOut;
      $dataOut = "<reportData  title=\"".$this->getDcTitle()."\" by=\"".$_SESSION["user"]["userName"]."\" on=\"".date(DATE_FORMAT)."\">".$dataOut."</reportData>";
    }
    $this->beginExport();
    print $dataOut;
    $this->endExport();
} /* end function doExport  */

public function fetchRecord() {
    $this->record["SERIAL_"] = $this->getRequestParam("SERIAL_");
    if (empty($this->record["SERIAL_"])) { throw new Exception("Missing value for primary key field SERIAL_ in DC0091.fetchRecord().");}
    $this->record["SID"] = $this->getRequestParam("SID");
    if (empty($this->record["SID"])) { throw new Exception("Missing value for primary key field SID in DC0091.fetchRecord().");}
    $pkArray = array("SERIAL_" => $this->record["SERIAL_"],"SID" => $this->record["SID"]);
    $this->findByPk($pkArray, $this->record);
    print "{success:true, data:".json_encode($this->record)."}";
} /* end function fetchRecord */

private function findByPk(&$pkCols, &$record) {
    $sql = "select 
                t.CLIENT_INFO
                ,t.LOGON_TIME
                ,t.MACHINE
                ,t.MODULE
                ,t.OSUSER
                ,t.PROCESS
                ,t.PROGRAM
                ,t.SADDR
                ,t.SCHEMANAME
                ,t.SERIAL_
                ,t.SERVER
                ,t.SERVICE_NAME
                ,t.SID
                ,t.SQL_ADDRESS
                ,t.SQL_ID
                ,t.SQL_TEXT
                ,t.STATE
                ,t.STATUS
                ,t.TERMINAL
                ,t.TYPE
                ,t.USERNAME
            from V_ORCL_SESSION t
         where 
           t.SERIAL_= :SERIAL_
           t.SID= :SID
            ";
    $rs = $this->getDbConn()->Execute($sql, $pkCols);
    $row = $rs->FetchRow();
    CollectionUtils::copyMapValues($record, $row);
} /* end function findByPk  */


public function callProcedure($pName) {
  $this->logger->debug("Start: ".$this->dcName.".callProcedure(".$pName.")");
    $this->populateRecordFromRequest();
    print "{success:true, data:".json_encode($this->record)."}";
    $this->logger->debug("End: ".$this->dcName.".callProcedure");
} /* end function callProcedure */


private function _initFields() {
  $this->fields = array(
  "CLIENT_INFO" => array(parent::FLDPROP_DATA_TYPE => "STRING")
  ,"LOGON_TIME" => array(parent::FLDPROP_DATA_TYPE => "STRING")
  ,"MACHINE" => array(parent::FLDPROP_DATA_TYPE => "STRING")
  ,"MODULE" => array(parent::FLDPROP_DATA_TYPE => "STRING")
  ,"OSUSER" => array(parent::FLDPROP_DATA_TYPE => "STRING")
  ,"PROCESS" => array(parent::FLDPROP_DATA_TYPE => "STRING")
  ,"PROGRAM" => array(parent::FLDPROP_DATA_TYPE => "STRING")
  ,"SADDR" => array(parent::FLDPROP_DATA_TYPE => "STRING")
  ,"SCHEMANAME" => array(parent::FLDPROP_DATA_TYPE => "STRING")
  ,"SERIAL_" => array(parent::FLDPROP_DATA_TYPE => "STRING")
  ,"SERVER" => array(parent::FLDPROP_DATA_TYPE => "STRING")
  ,"SERVICE_NAME" => array(parent::FLDPROP_DATA_TYPE => "STRING")
  ,"SID" => array(parent::FLDPROP_DATA_TYPE => "STRING")
  ,"SQL_ADDRESS" => array(parent::FLDPROP_DATA_TYPE => "STRING")
  ,"SQL_ID" => array(parent::FLDPROP_DATA_TYPE => "STRING")
  ,"SQL_TEXT" => array(parent::FLDPROP_DATA_TYPE => "STRING")
  ,"STATE" => array(parent::FLDPROP_DATA_TYPE => "STRING")
  ,"STATUS" => array(parent::FLDPROP_DATA_TYPE => "STRING")
  ,"TERMINAL" => array(parent::FLDPROP_DATA_TYPE => "STRING")
  ,"TYPE" => array(parent::FLDPROP_DATA_TYPE => "STRING")
  ,"USERNAME" => array(parent::FLDPROP_DATA_TYPE => "STRING")
);
}

}
?>

File: trunk/ServerJava/src/net/nan21/ebs/dc/DC0091.java
/* N21 eBusiness Suite
 * Copyright: Nan21 Electronics srl
 * Generated content.
 * DC0091 DC Controller: Oracle sessions
 */

package net.nan21.ebs.dc;


import java.util.*;
import javax.servlet.http.HttpServletResponse;
import net.nan21.lib.*;
import net.nan21.lib.dc.*;

public class DC0091 extends AbstractDataControl implements IDataControl {

  public void init (HttpRequest request, HttpServletResponse response, HttpSession session, DbManager dbm) throws Exception {
    this._initFields();
    super.init(request, response, session, dbm);
  }

private void preQuery() {
    this.queryWhere.append("type='USER'");
    if (this.request.getParam("QRY_SQL_TEXT") != null && !this.request.getParam("QRY_SQL_TEXT").equals("")) {
      this.queryWhere.append(( this.queryWhere.length() > 0 )?" and ":"");
      this.queryWhere.append("t.SQL_TEXT like :SQL_TEXT");
      this.queryParams.put("SQL_TEXT",(String)this.request.getParam("QRY_SQL_TEXT"));
    }
}

public void doQuery() throws Exception {
    this.prepareQueryContext();
    this.preQuery();
    this.queryWhere.insert(0, (this.queryWhere.length()>0)?" where ":"");
    String sql = "select "+ 
               " t.CLIENT_INFO"+
               " ,t.LOGON_TIME"+
               " ,t.MACHINE"+
               " ,t.MODULE"+
               " ,t.OSUSER"+
               " ,t.PROCESS"+
               " ,t.PROGRAM"+
               " ,t.SADDR"+
               " ,t.SCHEMANAME"+
               " ,t.SERIAL_"+
               " ,t.SERVER"+
               " ,t.SERVICE_NAME"+
               " ,t.SID"+
               " ,t.SQL_ADDRESS"+
               " ,t.SQL_ID"+
               " ,t.SQL_TEXT"+
               " ,t.STATE"+
               " ,t.STATUS"+
               " ,t.TERMINAL"+
               " ,t.TYPE"+
               " ,t.USERNAME"+
           " from V_ORCL_SESSION t "+this.queryWhere.toString()+" "+this.queryOrderBy;
    this.writeResultDoQuery(sql);
} 


public void doExport() throws Exception {
    this.prepareQueryContext();
    this.preQuery();
    this.queryWhere.insert(0, (this.queryWhere.length()>0)?" where ":"");
    String sql = "select "+ 
               " t.SID"+
               " ,t.SERIAL_"+
               " ,t.USERNAME"+
               " ,t.STATUS"+
               " ,t.MACHINE"+
               " ,t.TERMINAL"+
               " ,t.OSUSER"+
               " ,t.PROGRAM"+
               " ,t.TYPE"+
               " ,t.LOGON_TIME"+
               " ,t.SADDR"+
               " ,t.SERVER"+
               " ,t.SCHEMANAME"+
               " ,t.PROCESS"+
               " ,t.SQL_ADDRESS"+
               " ,t.SQL_ID"+
               " ,t.MODULE"+
               " ,t.CLIENT_INFO"+
               " ,t.STATE"+
               " ,t.SERVICE_NAME"+
               " ,t.SQL_TEXT"+
           " from V_ORCL_SESSION t "+this.queryWhere.toString()+" "+this.queryOrderBy;
    this.writeResultDoExport(sql);
}

 	public void fetchRecord()  throws Exception  {
     this.populateRecordPkFromRequest();
     this.findByPk();
       this.writeResultFetchRecord();	 
   }

public void doInsert() throws Exception  {}
public void doUpdate() throws Exception  {}
public void doDelete() throws Exception {}
public void initNewRecord() throws Exception {}
private void findByPk()  throws Exception {
    String sql = "select "+ 
               " t.CLIENT_INFO"+
               " ,t.LOGON_TIME"+
               " ,t.MACHINE"+
               " ,t.MODULE"+
               " ,t.OSUSER"+
               " ,t.PROCESS"+
               " ,t.PROGRAM"+
               " ,t.SADDR"+
               " ,t.SCHEMANAME"+
               " ,t.SERIAL_"+
               " ,t.SERVER"+
               " ,t.SERVICE_NAME"+
               " ,t.SID"+
               " ,t.SQL_ADDRESS"+
               " ,t.SQL_ID"+
               " ,t.SQL_TEXT"+
               " ,t.STATE"+
               " ,t.STATUS"+
               " ,t.TERMINAL"+
               " ,t.TYPE"+
               " ,t.USERNAME"+
           " from V_ORCL_SESSION t"+
        " where "+
     "      t.SERIAL_= :SERIAL_"+ 
     "      t.SID= :SID"+ 
          "";
    this.record = dbm.executeQuery(sql, this.recordPk).get(0); 
} 


public void doCustomAction(String pName)  throws Exception {
    this.populateRecordFromRequest();
    this.sendRecord();
}


	private void  _initFields() {
	  this.fields = new HashMap<String, FieldDef>();
	  this.fields.put("CLIENT_INFO", new FieldDef("STRING"));
	  this.fields.put("LOGON_TIME", new FieldDef("STRING"));
	  this.fields.put("MACHINE", new FieldDef("STRING"));
	  this.fields.put("MODULE", new FieldDef("STRING"));
	  this.fields.put("OSUSER", new FieldDef("STRING"));
	  this.fields.put("PROCESS", new FieldDef("STRING"));
	  this.fields.put("PROGRAM", new FieldDef("STRING"));
	  this.fields.put("SADDR", new FieldDef("STRING"));
	  this.fields.put("SCHEMANAME", new FieldDef("STRING"));
	  this.fields.put("SERIAL_", new FieldDef("STRING"));
	  this.fields.put("SERVER", new FieldDef("STRING"));
	  this.fields.put("SERVICE_NAME", new FieldDef("STRING"));
	  this.fields.put("SID", new FieldDef("STRING"));
	  this.fields.put("SQL_ADDRESS", new FieldDef("STRING"));
	  this.fields.put("SQL_ID", new FieldDef("STRING"));
	  this.fields.put("SQL_TEXT", new FieldDef("STRING"));
	  this.fields.put("STATE", new FieldDef("STRING"));
	  this.fields.put("STATUS", new FieldDef("STRING"));
	  this.fields.put("TERMINAL", new FieldDef("STRING"));
	  this.fields.put("TYPE", new FieldDef("STRING"));
	  this.fields.put("USERNAME", new FieldDef("STRING"));
	  String[] _pkFields = {"SERIAL_","SID"};
	  this.pkFields = _pkFields;
	  String[] _summaryFields = {};
	  this.summaryFields = _summaryFields;
	  this.queryResultSize = 40;
	}

}
